gensim/test/test_translation_matrix.py

```diff
@@ -0,0 +1,118 @@
+        test_source_word, test_target_word = zip(*test_word_pair)
+        translated_words = model.translate(test_source_word, topn=3)
+
+        self.assertTrue("uno" in translated_words["one"])
+        self.assertTrue("due" in translated_words["two"])
+        self.assertTrue("mela" in translated_words["apple"])
+        self.assertTrue("arancione" in translated_words["orange"])
+        self.assertTrue("cane" in translated_words["dog"])
+        self.assertTrue("maiale" in translated_words["pig"])
+        self.assertTrue("gatto" in translated_words["cat"])
+
+    def test_translate_gc(self):
+        # test globally corrected neighbour retrieval method
+        model = translation_matrix.TranslationMatrix(self.word_pair, self.source_word_vec, self.target_word_vec)
+        model.train(self.word_pair)
+
+        test_word_pair = [("one", "uno"), ("two", "due"), ("apple", "mela"), ("orange", "aranicione"), ("dog", "cane"), ("pig", "maiale"), ("cat", "gatto")]
+        test_source_word, test_target_word = zip(*test_word_pair)
+        translated_words = model.translate(test_source_word, topn=3, gc=1, sample_num=10)
+
+        self.assertTrue("uno" in translated_words["one"])
+        self.assertTrue("due" in translated_words["two"])
+        self.assertTrue("mela" in translated_words["apple"])
+        self.assertTrue("arancione" in translated_words["orange"])
+        self.assertTrue("cane" in translated_words["dog"])
+        self.assertTrue("maiale" in translated_words["pig"])
+        self.assertTrue("gatto" in translated_words["cat"])
+
+
+def read_sentiment_docs(filename):
+    sentiment_document = namedtuple('SentimentDocument', 'words tags')
+    alldocs = []  # will hold all docs in original order
+    with gensim.utils.smart_open(filename, encoding='utf-8') as alldata:
+        for line_no, line in enumerate(alldata):
+            tokens = gensim.utils.to_unicode(line).split()
+            words = tokens
+            tags = str(line_no)
+            alldocs.append(sentiment_document(words, tags))
+    return alldocs
+
+
+class TestBackMappingTranslationMatrix(unittest.TestCase):
+    def setUp(self):
+        filename = datapath("alldata-id-10.txt")
+        train_docs = read_sentiment_docs(filename)
+        self.train_docs = train_docs
+        self.source_doc_vec_file = datapath("small_tag_doc_5_iter50")
+        self.target_doc_vec_file = datapath("large_tag_doc_10_iter50")
+
+        self.source_doc_vec = Doc2Vec.load(self.source_doc_vec_file)
+        self.target_doc_vec = Doc2Vec.load(self.target_doc_vec_file)
+
+    def test_translation_matrix(self):
+        model = translation_matrix.BackMappingTranslationMatrix(self.train_docs[:5], self.source_doc_vec, self.target_doc_vec)
+        transmat = model.train(self.train_docs[:5])
+        self.assertEqual(transmat.shape, (100, 100))
```

Comment:
Why only "sanity check"? Please add another tests (similar with translation tests from previous class)?

Reply:
This is a little different with the translation of word. We build the mapping `F` between the target model and original model(old). Given a document vector `x` from the target model,  we infer the vector by multiple `F` with `x` to get a new vector for the original vector. By now, we didn't have a good evaluation metric to assess the vector. So I only give the "sanity check" now.

Reply:
You can calculate cosine distance between vector from model and vector from translation